src/internal/wasm/wasm_init_data.c

```diff
@@ -0,0 +1,26 @@
+#include <wasm/wasm_init_data.h>
+#include <bits/hwcap.h>
+#include <bits/limits.h>
+#include <elf.h>
+
+static const char* const progname = "/a.out";
+
+struct wasm_init_data_t wasm_init_data = {
+	.argc = 1,
+	.argv0 = progname,
+	.argv1 = 0,
+	.envp0 = 0,
+	.aux = {
+		// It would be nice if Wasm provided AT_RANDOM, oh well.  The rest
+		// can just be dummies.
+		AT_HWCAP,    HWCAP_MVP,
+		AT_PAGESZ,   PAGE_SIZE,
+		AT_EXECFN,   (uintptr_t)progname,
+#if defined __wasm32__
+		AT_PLATFORM, (uintptr_t)"wasm32",
+#elif defined __wasm64__
+		AT_PLATFORM, (uintptr_t)"wasm64",
+#endif
+		AT_NULL,     0
+	}
+};
```

Comment:
Random musing: I realize that this serves a purpose right now, though at some point we may want to explore other options. Portable code won't likely use interfaces like `getauxv` in the first place, for wasm-specific code there are better ways to obtain most of this information, and for musl initialization, there isn't all that much code that needs these, and some of it needs auxv fields that aren't supported (AT_PHDR etc.).